commit/merkleroot/query.go

```diff
@@ -0,0 +1,61 @@
+package merkleroot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink-ccip/commit/merkleroot/rmn"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+)
+
+func (w *Processor) Query(ctx context.Context, prevOutcome Outcome) (Query, error) {
+	if !w.cfg.RMNEnabled {
+		return Query{}, nil
+	}
+
+	nextState := prevOutcome.NextState()
+	if nextState != BuildingReport {
+		return Query{}, nil
+	}
+
+	offRampAddress, err := w.ccipReader.GetContractAddress(consts.ContractNameOffRamp, w.cfg.DestChain)
+	if err != nil {
+		return Query{}, fmt.Errorf("get offRamp contract address: %w", err)
+	}
+	dstChainInfo := rmn.DestChainInfo{
+		Chain:          w.cfg.DestChain,
+		OffRampAddress: offRampAddress,
+	}
+
+	reqUpdates := make([]rmn.FixedDestLaneUpdateRequest, 0, len(prevOutcome.RangesSelectedForReport))
```

Comment:
Could this be a map so that uniqueness is clear?
```suggestion
	reqUpdates := make(map[ChainSelector]rmn.FixedDestLaneUpdateRequest, 0, len(prevOutcome.RangesSelectedForReport))
```

Reply:
It's similar to the shared proto types, so better keep it consistent with rmn types. Also the types will be replaced with the auto-generated ones.

Reply:
I'm not suggesting you modify `FixedDestLaneUpdateRequest`.

It needs to be re-encapsulated in an `ObservationRequest` anyway, so this slice needs some processing either way:
```
message ObservationRequest {
  LaneDest lane_dest = 1; // could be implied
  repeated FixedDestLaneUpdateRequest fixed_dest_lane_update_requests = 2;
}
```

Putting it in a map avoids a category of bugs so it seems worthwhile.

Reply:
Oh I see, yeah but maybe in the future we actually want to pass multiple ranges for the same chain since RMN supports it, so IMO we shouldn't treat it as unique.